src/screens/transfer/screen/delegateScreen.js

```diff
@@ -101,6 +171,63 @@ class DelegateScreen extends Component {
     this.setState({ from: value, amount: 0 });
   };
 
+  _handleSliderValueChange = (value, availableVestingShares) => {
+    const { hivePerMVests } = this.props;
+    if (value === availableVestingShares) {
+      this.setState({ isAmountValid: false });
+    }
+    if (value !== availableVestingShares) {
+      this.setState({
+        step: 2,
+        isAmountValid: true,
+        amount: value,
+        hp: vestsToHp(value, hivePerMVests).toFixed(3),
+      });
+    } else {
+      this.setState({ amount: value, hp: vestsToHp(value, hivePerMVests).toFixed(3), step: 2 });
+    }
+  };
+
+  _handleNext = () => {
+    const { step, hp, amount, destination, from, delegatedHP } = this.state;
+    const { dispatch, intl } = this.props;
+    if (step === 1) {
+      // this.setState({ step: 2 });
+    } else {
+      // this.amountTextInput.current.blur();
+      dispatch(
+        showActionModal(
+          intl.formatMessage({ id: 'transfer.confirm' }),
+          intl.formatMessage(
```

Comment:
actually we have to show them conditionally and they have to pass through redux so  it would be very complex. para props seems a straigh forward approach and less complex. Also we can show it conditionally